pkg/query/flamegraph_arrow.go

```diff
@@ -173,9 +174,10 @@ func generateFlamegraphArrowRecord(ctx context.Context, mem memory.Allocator, tr
 	cumulative := int64(0)
 	// This keeps track of the max depth of our flame graph.
 	maxHeight := int32(0)
-	// This keeps track of the root rows.
+	// This keeps track of the root rows indexed by the labels string.
+	// If the stack trace has no labels, we use the empty string as the key.
 	// This will be the root row's children, which is always our row 0 in flame graphs.
-	rootsRow := []int{}
+	rootsRow := map[string][]int{}
```

Comment:
is there any other way other than a map that we could do this? profiling data already suggests that a pretty large amount of time is spent in map accesses

Reply:
We want this to be unique per pprof label JSON string. 
We want to be able to find the individual rows under a specific label string. 
I'm not really sure if we have some better primitives right now. We would need to investigate some other data structures altogether. 
Happy to do this as a follow-up with some more profiling/benchmarking data.

Reply:
Sounds good as a follow-up and based on data.